src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -912,6 +983,14 @@ public partial record HostUrl(string Url) : IValueProvider, IManifestExpressionP
         System.Threading.Tasks.ValueTask<string?> IValueProvider.GetValueAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
     }
 
+    public partial interface IAspireStore
+    {
+        string BasePath { get; }
+
+        string GetFileNameWithContent(string filenameTemplate, System.IO.Stream contentStream);
+        string GetFileNameWithContent(string filenameTemplate, string sourceFilename);
```

Comment:
Is this GetOrCreate? Why is the Stream passed in and not returned?

Reply:
cc @sebastienros

Reply:
The stream is used to be copied to a new file in the store. Honestly we could make things work with either a file or a stream in this interface, no opinion what's best.

Reply:
The point of this API is to get a deterministic file name given the contents of the file. So if the contents change, the file name changes.

What would be the point of returning the Stream? We've already read from it and wrote the contents to the file in the store.

Reply:
I see it's really `GetFileNameForContent`

Reply:
https://github.com/dotnet/aspire/pull/7626